src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorProjectEngine.cs

```diff
@@ -88,19 +88,21 @@ protected override RazorCodeDocument CreateCodeDocumentCore(RazorProjectItem pro
             return CreateCodeDocumentCore(sourceDocument, projectItem.FileKind, importSourceDocuments, tagHelpers: null, configureParser, configureCodeGeneration);
         }
 
-        protected RazorCodeDocument CreateCodeDocumentCore(
+        protected internal RazorCodeDocument CreateCodeDocumentCore(
```

Comment:
I need this to be able to create code documents correctly. Basically it's not super valid to create a code document from scratch unless you're going to call Process on it, but that's not the pattern we use in most tests.